contracts/GIVpower.sol

```diff
@@ -109,32 +109,32 @@ contract GIVpower is GardenUnipoolTokenDistributor, IERC20MetadataUpgradeable {
             revert CannotUnlockUntilRoundIsFinished();
         }
 
-        for (uint256 i = 0; i < accounts.length; i++) {
+        for (uint256 i = 0; i < accounts.length;) {
             address _account = accounts[i];
             UserLock storage _userLock = userLocks[_account];
             RoundBalance storage _roundBalance = _userLock.roundBalances[round];
 
             // @dev Based on the design, unlockableTokenAmount and releasablePowerAmount are both zero or both positive
-            if (_roundBalance.unlockableTokenAmount == 0) {
-                // && _roundBalance._releasablePowerAmount == 0
-                continue;
-            }
+            if (_roundBalance.unlockableTokenAmount > 0) {
+                uint256 _releasablePowerAmount = _roundBalance.releasablePowerAmount;
+                uint256 _unlockableTokenAmount = _roundBalance.unlockableTokenAmount;
 
-            uint256 _releasablePowerAmount = _roundBalance.releasablePowerAmount;
-            uint256 _unlockableTokenAmount = _roundBalance.unlockableTokenAmount;
+                _userLock.totalAmountLocked = _userLock.totalAmountLocked.sub(_unlockableTokenAmount);
```

Comment:
@KurtMerbeth Do you think it's worth and safe to make this one unchecked too?

Reply:
i don't see a situation where `_totalAmountLocked < _unlockableTokenAmount` is. So it should be secure and safe some gas

Reply:
Essentially `_userLock.totalAmountLocked` must be bigger than `_unlockableTokenAmount` if the rest of the code works correctly. But you are right, let's leave it for safety